### DISCUSSION TITLE
WithStartTime on ChangeFeedProcessorBuilder

### DISCUSSION THREAD

**Discussion by derek-andrew-govx** [IEvangelist/azure-cosmos-dotnet-repository#358] [Q&A] [answered]
Did I just miss how to configure WithStartTime on a change feed or is that not implemented? If not implemented, are there issues with it? If no issues, could that be added?

**Comment by mumby0168** [comment]
Hi @derek-andrew-govx what is your use case? Are you asking if there is any documentation around the use of WithStartTime or?

**Reply by derek-andrew-govx** [reply]
@mumby0168 The use case is were are moving all of our applications to be more event driven and part of that is using change feeds. So we are deploying multiple changes feed processors, many of which depend on others, at once.  We either need to be careful on the order of startup so the lease is setup correctly or use a start time so they can be deployed in any order and not miss anything. Technically that is only needed once but does make deployment easier. 

We also have had use cases where we needed change feeds on existing Cosmos containers for the data that existed and changes going forward. Again, only needed once.

I didn't find any documentation or see a way in the code to set this but if you know where that is, I would very much appreciate being pointed in the right direction.

**Reply by mumby0168** [reply, marked as answer]
This is the MS docs for it:

https://learn.microsoft.com/azure/cosmos-db/nosql/change-feed-processor?tabs=dotnet#starting-time

We don't expose this via the library at the minute, but it can be added to the options class here to allow this, see the links below:

https://github.com/IEvangelist/azure-cosmos-dotnet-repository/blob/main/src/Microsoft.Azure.CosmosRepository/ChangeFeed/ChangeFeedOptions.cs

Then it would need to be added here if set:

https://github.com/IEvangelist/azure-cosmos-dotnet-repository/blob/7430997c2d62ce9f6e0c26f124ca16bc4cd5d56a/src/Microsoft.Azure.CosmosRepository/ChangeFeed/DefaultConatinerChangeFeedProcessor.cs#L40C8-L50C10

**Reply by derek-andrew-govx** [reply]
@mumby0168 Well, that makes me feel better. I didn't miss it and my local implementation is pretty much what you laid out above. If I figure out a way to unit test it, I may open a PR.

Thanks!
